src/jl_exported_funcs.inc

```diff
@@ -152,6 +152,7 @@
     XX(jl_gc_add_finalizer_th) \
     XX(jl_gc_add_ptr_finalizer) \
     XX(jl_gc_add_quiescent) \
+    XX(jl_gc_alloc) \
```

Comment:
@staticfloat @vtjnash any reason why this wasn't included before?

Reply:
not especially, though we try to make the API larger than necessary (this is not in src/julia.h for example), they get named mangled which is mildly annoying for debugging, and it is only needed for external C clients, not sysimg linking (which already gets direct access to all of libjulia-internal)